app/main.py

```diff
@@ -1,3 +1,46 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+
+        self.km += other
+        return self
+
+    def __mul__(self, other: Distance | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Distance | float) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Distance | float) -> bool:
+        return self.km < other
```

Comment:
why don't you check the type of `other`? It will not work correctly for Distance instance

Reply:
Still not fixed. You should check the type of other, and implement the correct comparison. The same in other similar cases

Reply:
Not fixed...
`if isinstance(other, Distance)` - here you check if other is a type of `Distance`. You have to provide such logic here and in other similar methods.